storage/src/lsm_tree/components/sstable.rs

```diff
@@ -192,8 +193,8 @@ impl SstableMeta {
         //     self.block_metas.last().as_ref().unwrap().last_key
         // );
 
-        !(self.block_metas.first().as_ref().unwrap().first_key > range.end()
-            || self.block_metas.last().as_ref().unwrap().last_key < range.start())
+        !(&&self.block_metas.first().as_ref().unwrap().first_key[..] > range.end()
+            || &&self.block_metas.last().as_ref().unwrap().last_key[..] < range.start())
```

Comment:
It's so confused with `&&` here.

Reply:
how about change it to `&self.block_metas.last().as_ref().unwrap().last_key[..] < *range.start()`?

Reply:
Better.